### PR TITLE
TropiCataclysm 0.6: Mounds and Blossoms

### PR DESCRIPTION
**Pull request by Xaritscin**
#### Summary
Mods "Adds Termites and new content for Triffids"

#### Purpose of change

First foray of TropiCataclysm into stuff related to Triffids, one of the main players in the setting. The PR also includes some new giant insects to decorate the region, including Termites.

#### Describe the solution

PR so far includes moths and butterflies in its base draft, will add the rest of the stuff as time goes on but the main list should be:

-add lepidopterans(butterflies, moths, and their larva) **DONE**
-add termites with their own feel different from ants **DONE**
-add termite mounds to the overmap **DONE**
-add new triffid locations with new enemies inside **DONE**

#### Describe alternatives you've considered

divide this PR into smaller parts like i have been doing and work one thing at a time.

#### Testing

Phase 1: Checked for astyle and jsonstyle validations then ran tests on my machine. fixed several crashes related to syntax not noticed by the system. spawned some termites to check their behavior, special attacks and aggressive response seem to be working as intended.

Phase 2: Opened a new map, received a few crashes at loading due to not having mapgen elements for certain mound features like the shaft, chimney and reservoir. game continued loading and was able to look around the map for a generated termite mound. all expected spawns and structure were working as intended (for this prototype). further work is required to refine it.

I also re-tested the termite nasute attack but im fiding that the player has to be really close to get aggroed by the termites so cannot test the full range of the special attack because the termites tun neutral again.

Phase 3: Reloaded map with the latest changes, character was able to find a mound with its new walling and check on the new fungal chambers and improved royal chambers. harvested some mutant fungus in winter without much effort. only error was crashes related to invalid ID for roofs in Mutable Overmaps. opened an issue for that.

Phaser 4: Reloaded map with the last locatiosn to check if they generated. roughly same chance as Fungal Towers and Blooms so they arent too common and the enemies were spawning alright.

#### Additional context

not sure how long im gonna be with this PR so i dont expect to add more updates to mod while working on this. 

PR made a small addition to special_locations.json because mutable overmaps currently dont have an ID avaliable for zlvls above the surface ground. it didnt get used in the end but may be useful for other projects in mainline.

### CODE REVIEW

File: data/mods/TropiCataclysm/monsters/tropical_insect_spider.json [hunk 1]
```diff
@@ -41,47 +41,132 @@
...
     "type": "MONSTER",
-    "name": { "str": "horned beetle" },
-    "description": "A large horned beetle of undescribed kind.",
-    "copy-from": "mon_heraclean_beetle",
-    "proportional": { "hp": 0.5, "speed": 1.1, "morale": 0.67, "melee_dice_sides": 0.67 },
-    "volume": "40 L",
-    "weight": "45 kg",
-    "upgrades": { "half_life": 14, "into": "mon_heraclean_beetle" }
+    "name": { "str": "mutated butterfly", "str_pl": "mutated butterflies" },
+    "description": "An oversized butterfly with vibrant color markings.  Unlike the other big insects of the cataclysm this is a very peacefull nectivore.",
```
Comment (NetSysFire) [inline data/mods/TropiCataclysm/monsters/tropical_insect_spider.json:47]:
```suggestion
    "description": "An oversized butterfly with vibrant colored markings.  Unlike the other big insects of the Cataclysm this is a very peaceful nectivore.",
```

File: data/mods/TropiCataclysm/monsters/tropical_insect_spider.json [hunk 2]
```diff
@@ -41,47 +41,132 @@
...
   },
   {
-    "id": "mon_heraclean_beetle",
+    "id": "mon_butterfly_giant",
     "type": "MONSTER",
-    "name": { "str": "heraclean beetle" },
-    "description": "A huge beetle the size of cattle.  Doesn't look predatory but the thick carapace and the enormous horn on its head seem like it can be formidable when fighting for mates.",
-    "default_faction": "beetle",
+    "name": { "str": "giant butterfly", "str_pl": "giant butterflies" },
+    "description": "A mutated butterfly larger than a human.  Despite its size is very lightweight and feeds mainly from the nectar produced by triffid flowers.",
```
Comment (NetSysFire) [inline data/mods/TropiCataclysm/monsters/tropical_insect_spider.json:71]:
```suggestion
    "description": "A mutated butterfly larger than a human.  Despite its size it is very lightweight and feeds mainly on the nectar produced by triffid flowers.",
```

File: data/mods/TropiCataclysm/monsters/tropical_insect_spider.json [hunk 3]
```diff
@@ -41,47 +41,132 @@
...
+    "reproduction": { "baby_egg": "egg_butterfly", "baby_count": 1, "baby_timer": 15 },
+    "baby_flags": [ "SPRING", "SUMMER" ],
+    "fear_triggers": [ "HURT", "FIRE" ],
+    "flags": [ "SEES", "HEARS", "FLIES", "STUMBLES" ]
+  },
+  {
+    "id": "mon_moth",
+    "type": "MONSTER",
+    "name": { "str": "mutated moth" },
+    "description": "An oversized moth with its typical wooden motiffs.  Its as peacefull as the mutated butterflies found in the day.",
```
Comment (NetSysFire) [inline data/mods/TropiCataclysm/monsters/tropical_insect_spider.json:94]:
```suggestion
    "description": "An oversized moth with its typical wood-like motifs.  It is as peaceful as the mutated butterflies found during the day.",
```

File: data/mods/TropiCataclysm/monsters/tropical_insect_spider.json [hunk 4]
```diff
@@ -41,47 +41,132 @@
...
+    "reproduction": { "baby_egg": "egg_butterfly", "baby_count": 1, "baby_timer": 15 },
+    "baby_flags": [ "SPRING", "SUMMER" ],
+    "fear_triggers": [ "HURT", "FIRE" ],
+    "flags": [ "SEES", "HEARS", "FLIES", "STUMBLES" ]
+  },
+  {
+    "id": "mon_caterpillar",
+    "type": "MONSTER",
+    "name": { "str": "caterpillar" },
+    "description": "Caterpillars are the larval stage of butterflies and moths.  This one seems newly hatched from an egg but even then its as large as a cat.",
```
Comment (NetSysFire) [inline data/mods/TropiCataclysm/monsters/tropical_insect_spider.json:141]:
```suggestion
    "description": "Caterpillars are the larval stage of butterflies and moths.  This one seems newly hatched from an egg but even now it is as large as a cat.",
```

File: data/mods/TropiCataclysm/monsters/tropical_insect_spider.json [hunk 5]
```diff
@@ -41,47 +41,132 @@
...
+    "dodge": 2,
+    "harvest": "arachnid",
+    "upgrades": { "age_grow": 14, "into": "GROUP_BUTTERFLY" },
+    "flags": [ "SEES", "SMELLS", "CLIMBS", "HEARS", "STUMBLES", "PATH_AVOID_FIRE" ]
+  },
+  {
+    "id": "mon_caterpillar_giant",
+    "type": "MONSTER",
+    "name": { "str": "giant caterpillar" },
+    "description": "This caterpillar has reached the size of a person and it looks close to get into the cocoon phase.  You can only imagine the size of the insect that will come out of it.",
```
Comment (NetSysFire) [inline data/mods/TropiCataclysm/monsters/tropical_insect_spider.json:163]:
```suggestion
    "description": "This caterpillar has reached the size of a person and it looks close to reaching the pupa life stage.  You can only imagine the size of the insect that will emerge.",
```

See https://en.wikipedia.org/wiki/Pupa, I reworded it a bit so it looks a bit more smooth.

**Comment by Xaritscin** [comment]
added termites into the list with all their required workings for a proper generation. However, there's still no mound overmap special to put them into the game.

I also have to look into the shooter soldier special attacks. according to Wikipedia certain species of termites have a second class of soldier which shoots chemicals but unlike acid ants i want to go for something like sludge-bile to slow down the player.

File: data/mods/TropiCataclysm/monsters/tropical_insect_spider.json [hunk 6]
```diff
@@ -41,47 +41,346 @@
...
+    "vision_night": 5,
+    "harvest": "arachnid",
+    "anger_triggers": [ "FRIEND_ATTACKED", "FRIEND_DIED", "HURT", "PLAYER_CLOSE" ],
+    "flags": [ "SEES", "HEARS", "SMELLS", "CLIMBS", "PATH_AVOID_FIRE", "PATH_AVOID_FALL" ]
+  },
+  {
+    "id": "mon_termite_alate",
+    "type": "MONSTER",
+    "name": { "str": "termite alate" },
+    "description": "An enormous winged termite dedicated to spreading new colonies.  They arent violent but will respond to aggresion if molested.",
```
Comment (BrettDong) [inline data/mods/TropiCataclysm/monsters/tropical_insect_spider.json:303]:
```suggestion
    "description": "An enormous winged termite dedicated to spreading new colonies.  They are not violent but will respond to aggression if molested.",
```

**Comment by Xaritscin** [comment]
Started the implementation of Termite Mounds into the mod. as a starting prototype i use a modified version of the vanilla Anthills. will wait for validation errores before testing.

However, the idea isnt to just copy the shape of an anthill. a Termite Mound is a mixed structure that is both OVER and UNDER the ground so i need to figure out how to creathe the main entrances, the shaft and the chimneys. the underground parts should be similar in a way to Anthills but it should probably covered a lot more Z-lvls than this like in this image i found with google:

[https://images-na.ssl-images-amazon.com/images/I/915ZL7gHvzL.jpg](url)

**Comment by Xaritscin** [comment]
Started doing modifications to the base anthill structure. im not sure if im ever going to make something close to a real termite mound but for a starting part i made a dirt structure similar to the dermatik nest that can be entered from 2 sides. in reality there should be ant-like entrance tunnels outside while the main structure is a gigantic spire of dirt with a few capillary tunnels and a central shaft but my map makin skills are very basic.

so for now the mound is just a pile of dirt with two holes and a roof.

i also started doing work on the fungal farms. the chamber in itself is much like the rest of the spaces in the colony a copy of the food storage but its now layered on a new terrain i made based out of the fungal bed (but without the FUNGUS) field so it doesnt generate mycus stuff.

The idea is that the Termites and possibly the Ants have been cultivating a slightly mutated version of their regular terran fungus on their colonies. this fungus may be cultivated as if it was underbrush or a crop and is completely unrelated to mycus so it doesnt produce any kind of expansion.

**Comment by Xaritscin** [comment]
Mound seems to be fnished for the time being as current Mutable Overmap properties dont allow to make a proper roofing. will move onto the next phase of the PR which is Trifid locations. these should be less problematic as they are above ground structures  for the most part

**Comment by Xaritscin** [comment]
after messing around i have added a new "open_air" part to Special Locations on the vanilla files but i couldnt make it work for my structures on this PR. HOWEVER, it should prove handy for anyone wanting to make Mutable Overmaps which make use of varius Zlvls above ground. i will start setting up the Triffid part of my PR and leave the termites for the time being

**Comment by Xaritscin** [comment]
submitting the PR for merge

**Pull request merged**